From: Thomas Renninger <trenn@suse.de>
Subject: X86: Solve Dell Latitudes do not reboot on x86_64 more generally
References: none
Patch-Mainline: submitted

There are quite some reports that those do not reboot:
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/833705
->E6520, E6220
http://www.linux-club.de/viewtopic.php?f=29&t=114459&start=20
->E5520
and some more already are blacklisted

This patch improves the blacklist for all E6xxx and E5xxx Latitudes.

Strange is that on above Ubuntu bug it was mentioned that 32 bit
reboots just fine, which should use the same mechanism than 64 bit
by default (ACPI).

While this should solve the problem on short term for quite some users,
hopefully a more generic way is found sooner or later...


Signed-off-by: Thomas Renninger <trenn@suse.de>

---
 arch/x86/kernel/reboot.c |   20 ++++++--------------
 1 file changed, 6 insertions(+), 14 deletions(-)

--- a/arch/x86/kernel/reboot.c
+++ b/arch/x86/kernel/reboot.c
@@ -428,28 +428,20 @@ static struct dmi_system_id __initdata p
 			DMI_MATCH(DMI_PRODUCT_NAME, "iMac9,1"),
 		},
 	},
-	{	/* Handle problems with rebooting on the Latitude E6320. */
+	{
 		.callback = set_pci_reboot,
-		.ident = "Dell Latitude E6320",
+		.ident = "Dell Latitude E5xxx",
 		.matches = {
 			DMI_MATCH(DMI_SYS_VENDOR, "Dell Inc."),
-			DMI_MATCH(DMI_PRODUCT_NAME, "Latitude E6320"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Latitude E5"),
 		},
 	},
-	{	/* Handle problems with rebooting on the Latitude E5420. */
+	{
 		.callback = set_pci_reboot,
-		.ident = "Dell Latitude E5420",
+		.ident = "Dell Latitude E6xxx",
 		.matches = {
 			DMI_MATCH(DMI_SYS_VENDOR, "Dell Inc."),
-			DMI_MATCH(DMI_PRODUCT_NAME, "Latitude E5420"),
-		},
-	},
-	{	/* Handle problems with rebooting on the Latitude E6420. */
-		.callback = set_pci_reboot,
-		.ident = "Dell Latitude E6420",
-		.matches = {
-			DMI_MATCH(DMI_SYS_VENDOR, "Dell Inc."),
-			DMI_MATCH(DMI_PRODUCT_NAME, "Latitude E6420"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Latitude E6"),
 		},
 	},
 	{	/* Handle problems with rebooting on the OptiPlex 990. */
